app/models/YourDetails.scala

```diff
@@ -36,24 +36,25 @@ object YourDetails {
       "firstName" -> optional(text)
         .verifying("error.firstNameRequired", _.isDefined)
         .verifying("error.firstNameLength", i => i.isEmpty || i.getOrElse("").length <= 35)
-        .verifying("error.firstNamePattern", i => i.isEmpty || i.getOrElse("").length > 35 || i.getOrElse("").matches("""^[A-Za-z0-9 \-,.&'\\]{1,35}$""")),
+        .verifying("error.firstNamePattern", i => i.isEmpty || i.getOrElse("").length > 35 || i.getOrElse("").matches("""^[A-Za-z \-']{1,35}$""")),
 
       "lastName" -> optional(text)
         .verifying("error.lastNameRequired", _.isDefined)
         .verifying("error.lastNameLength", i => i.isEmpty || i.getOrElse("").length <= 35)
-        .verifying("error.lastNamePattern", i => i.isEmpty || i.getOrElse("").length > 35 || i.getOrElse("").matches("""^[A-Za-z0-9 \-,.&'\\]{1,35}$""")),
+        .verifying("error.lastNamePattern", i => i.isEmpty || i.getOrElse("").length > 35 || i.getOrElse("").matches("""^[A-Za-z \-']{1,35}$""")),
 
       "role" -> optional(text)
         .verifying("error.roleRequired", _.isDefined)
         .verifying("error.roleLength", i => i.isEmpty || i.getOrElse("").length <= 30)
-        .verifying("error.rolePattern", i => i.isEmpty || i.getOrElse("").length > 30 || i.getOrElse("").matches("""^[A-Za-z0-9 \-,.&'\/]{1,30}$""")),
+        .verifying("error.rolePattern", i => i.isEmpty || i.getOrElse("").length > 30 || i.getOrElse("").matches("""^[A-Za-z \-']{1,30}$""")),
 
       "phone" -> optional(text)
         .verifying("error.phoneRequired", _.isDefined)
         .verifying("error.phonePattern", i => i.isEmpty || i.getOrElse("").matches("""^[A-Z0-9 \)\/\(\*\#\-\+]{1,24}$""")),
 
-      "email" -> optional(email)
+      "email" -> optional(text)
         .verifying("error.emailRequired", _.isDefined)
+        .verifying("error.email", i => i.isEmpty || i.getOrElse("").matches("""(?:[A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[A-Za-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[A-Za-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])"""))
```

Comment:
beast of a regex, where did that come from 😎 (works just curious lol)

Reply:
Yeah, it's a long one! I got it from http://emailregex.com/ but had to add a bit to allow for capitals before the "@"